src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1972,6 +1972,11 @@ public void Paste(Point2D targetPoint, bool useOffset = true)
             var newNoteModels = new List<NoteModel>();
             foreach (var note in notes)
             {
+                // TODO: find out the issue that duplicates NoteModel with same GUID
+                // check if NoteModel was already inserted
+                if (modelLookup.ContainsKey(note.GUID))
```

Comment:
This effectively stops copy paste of notes?
If thats the case, I think we must actually fix the bug before we can merge this.

Reply:
Ok, but I have no idea where is the bug.